### PR TITLE
feat: DIA-1018: Output streaming

### PR DESCRIPTION
**Pull request by hakan458**

### CODE REVIEW

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/0f9249258fd131a239fca44c9ee5a35a09345123 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/4b57cd0aade01fb644e90540b4f8760e4914b2f4 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/cb0d15f7fe1acd7ab007c2ed0d0a100e3b3cfbe5 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/ee1b6e9ca1a7ee69dfca8522851ba81885043f1d is created.

File: adala/environments/kafka.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class AsyncKafkaEnvironment(AsyncEnvironment):
     kafka_input_topic: str
     kafka_output_topic: str
 
-    async def message_receiver(self, consumer: AIOKafkaConsumer, timeout: int = 3):
+    async def message_receiver(self, consumer: AIOKafkaConsumer, timeout: int = 30):
```
Comment (matt-bernstein) [inline adala/environments/kafka.py:35]: should probably revert this

File: server/app.py [hunk 2]
```diff
@@ -204,10 +210,17 @@ async def submit_streaming(request: SubmitStreamingRequest):
...
+    logger.info(f"Task {task.name} submitted with job_id {input_job_id}")
 
-    return Response[JobCreated](data=JobCreated(job_id=result.id))
+    task = process_streaming_output
+    logger.info(f"Submitting task {task.name}")
+    output_result = task.delay(job_id=input_job_id, result_handler=request.result_handler)
+    output_job_id = output_result.id
+    logger.info(f"Task {task.name} submitted with job_id {output_job_id}")
+
+    return Response[JobCreated](data=JobCreated(job_id=input_job_id))
```
Comment (pakelley) [inline server/app.py:223]: I know we didn't quite resolve this, but I think we should be returning `output_job_id` here, right?

File: server/tasks/process_file.py [hunk 3]
```diff
@@ -38,3 +42,58 @@ def process_file_streaming(self, serialized_agent: bytes):
 
     # Run the agent
     asyncio.run(agent.arun())
+
+
+async def async_process_streaming_output(input_job_id: str, result_handler: str, batch_size: int):
+    logger.info(f"Polling for results {input_job_id=}")
+
+    try:
+        result_handler = ResultHandler.__dict__[result_handler]
```
Comment (pakelley) [inline server/tasks/process_file.py:51]: nit: you should be able to just do `ResultHandler[result_handler]` here. Just tested in ipython, and it worked.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/63b53dc55a57d12779f8c67852ece6453cdf5f6e is created.

File: server/app.py (hunk 2 above)
Comment (hakan458) [reply server/app.py:223]: As of now definitely not , we need this job ID on LSE side to submit batches to the correct Kafka queue

File: server/tasks/process_file.py (hunk 3 above)
Comment (hakan458) [reply server/tasks/process_file.py:51]: No, it doesnt when its a function in the enum. Try it

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/7342a5b5ed4ea1a980ed96a12751426babd17dfb is created.

**Review by forum-hs — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/9c736305275418ac4404af96325177dda2e44c29 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/bd643eba845bf8d8988f5205f69689f624354def is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/f8d36ee015b21b27f5319470580450a2a786e121 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/e6a6920d98d322173db3d7a365e5da5da2d30767 is created.

**Pull request merged**